### PR TITLE
Debug Message Improvements

### PR DESCRIPTION
**Pull request by francinum**
## About The Pull Request

Unifies debug messaging as much as reasonable.

## Why It's Good For The Game

spans fancy.

## Changelog

:cl:
admin: Live Debugging is now a bit more verbose.
/:cl:

### CODE REVIEW

File: code/__HELPERS/debug_messaging.dm [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		if(DBG_ERROR)
+			severity_class = "debug_error"
+		else
+			message_debug(DBG_ERROR, "DebugMessage/Severity", "Invalid severity level [level]. Setting to error.", DBG_STACK_TRACE)
+			level = DBG_ERROR
+			severity_class = "debug_error"
+
+	// Check if we throw a stack trace.
+	if(additional & DBG_STACK_TRACE)
+		stack_trace("[level]| [built_message]")
```
Comment (Kapu1178) [inline code/__HELPERS/debug_messaging.dm:63]:
```suggestion
		stack_trace("[level] | [built_message]")
```

Comment (Kapu1178) [inline code/__HELPERS/debug_messaging.dm:63]:
```suggestion
		stack_trace("[level] | [built_message]")
```

File: code/__HELPERS/debug_messaging.dm [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	// Log to world.
+	if(additional & DBG_LOG_WORLD)
+		log_world("[level]| [built_message]")
```
Comment (Kapu1178) [inline code/__HELPERS/debug_messaging.dm:66]:
```suggestion
		log_world("[level] | [built_message]")
```

File: code/controllers/master.dm [hunk 3]
```diff
@@ -248,7 +248,7 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 			current_initializing_subsystem = subsystem
 
 			if(GLOB.is_debug_server)
-				to_chat(world, span_boldnotice("Initializing [subsystem.name]..."))
```
Comment (Kapu1178) [inline code/controllers/master.dm:251]: Don't make these debug spans, I like the way it looks as-is.

File: code/controllers/subsystem.dm [hunk 4]
```diff
@@ -277,8 +277,9 @@
 	var/time = (REALTIMEOFDAY - start_timeofday) / 10
 	var/msg = "Initialized [name] subsystem within [time] second[time == 1 ? "" : "s"]!"
 	if(GLOB.is_debug_server)
-		to_chat(world, span_debug("[msg]"))
-	log_world(msg)
+		message_debug(DBG_INFO, "MC/Initialize", msg)
+	else
+		log_world(msg)
```
Comment (Kapu1178) [inline code/controllers/subsystem.dm:282]: Don't make these debug spans, I like the way it looks as-is.

File: code/controllers/subsystem/airmachines.dm [hunk 5]
```diff
@@ -25,13 +25,13 @@ SUBSYSTEM_DEF(airmachines)
...
+	to_chat(world, span_debug_info("Airmachines: Setting up atmospheric machinery..."))
 	setup_atmos_machinery()
-	to_chat(world, span_debug("Airmachines: Airmachine setup completed in [(REALTIMEOFDAY- starttime) / 10] seconds!"))
+	to_chat(world, span_debug_info("Airmachines: Airmachine setup completed in [(REALTIMEOFDAY- starttime) / 10] seconds!"))
 	starttime = REALTIMEOFDAY
-	to_chat(world, span_debug("Airmachines: Creating pipenets..."))
+	to_chat(world, span_debug_info("Airmachines: Creating pipenets..."))
 	setup_pipenets()
-	to_chat(world, span_debug("Airmachines: Pipenet creation completed in [(REALTIMEOFDAY- starttime) / 10] seconds!"))
+	to_chat(world, span_debug_info("Airmachines: Pipenet creation completed in [(REALTIMEOFDAY- starttime) / 10] seconds!"))
```
Comment (Kapu1178) [inline code/controllers/subsystem/airmachines.dm:34]: Don't make these debug spans, I like the way it looks as-is.

File: code/controllers/subsystem/mapping.dm [hunk 6]
```diff
@@ -294,7 +294,7 @@ Used by the AI doomsday and the self-destruct nuke.
 	z_list = SSmapping.z_list
 	multiz_levels = SSmapping.multiz_levels
 
-#define INIT_ANNOUNCE(X) to_chat(world, span_debug("[X]")); log_world(X)
+#define INIT_ANNOUNCE(X) to_chat(world, X); log_world(X)
```
Comment (Kapu1178) [inline code/controllers/subsystem/mapping.dm:297]: Don't make these debug spans, I like the way it looks as-is.

File: tgui/packages/tgui-panel/styles/goon/chat-dark.scss [hunk 7]
```diff
@@ -980,7 +980,45 @@ em {
 
 .debug {
   color: #5ce43d;
+	background-color: black;
   font-weight: bold;
+	font-family: monospace;
+	border: 1px groove grey;
+	width: 80ch;
+	margin-top: 1px;
+	margin-bottom: 1px;
```
Comment (Kapu1178) [inline tgui/packages/tgui-panel/styles/goon/chat-dark.scss:989]:
```suggestion
	padding: 1px 4px 1px 4px;
	margin: 0 2px 0 0;
```

File: tgui/packages/tgui-panel/styles/goon/chat-dark.scss [hunk 8]
```diff
@@ -980,7 +980,45 @@ em {
...
+  color:orange;
+}
+
+.debug_error {
+	border-color: red;
+}
+.debug_error::before {
+  content: "ERROR|";
+  color:red;
 }
```
Comment (Kapu1178) [inline tgui/packages/tgui-panel/styles/goon/chat-dark.scss:1022]: Not really sure I like using content modifiers in CSS, as we don't really do it elsewhere. Don't care enough to block over it though.

```suggestion
.debug_trace{
  border-color: grey;
}
.debug_trace::before {
  content: "TRACE |";
  color:grey;
}

.debug_info {
	border-color: cyan;
}
.debug_info::before {
  content: "INFO |";
  color:cyan;
}

.debug_warn {
	border-color: orange;
}
.debug_warn::before {
  content: "WARN |";
  color:orange;
}

.debug_error {
	border-color: red;
}
.debug_error::before {
  content: "ERROR |";
  color:red;
}
```

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem.dm (hunk 4 above)
Comment (francinum) [reply code/controllers/subsystem.dm:282]: Unavoidable without manually styling, as this used the old debug span.

File: code/controllers/subsystem/airmachines.dm (hunk 5 above)
Comment (francinum) [reply code/controllers/subsystem/airmachines.dm:34]: Unavoidable without manually styling, as this used the old debug span.

File: code/controllers/subsystem/mapping.dm (hunk 6 above)
Comment (francinum) [reply code/controllers/subsystem/mapping.dm:297]: This used the old debug span as well.

**Comment by Kapu1178** [comment]
the zas and airmachines text lack spans

File: tgui/packages/tgui-panel/styles/goon/chat-light.scss [hunk 9]
```diff
@@ -978,9 +978,52 @@ h1.alert, h2.alert {
   color: #2e2e2e;
 }
 
+.debugLegacy {
+  color: #5ce43d;
+  font-weight: bold;
+}
+
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply tgui/packages/tgui-panel/styles/goon/chat-light.scss:985]: @francinum

**Pull request closed without merging**